File: golang-app/pkg/model/task.entity.go
package model

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type TaskEntity struct {
	Id                  primitive.ObjectID `bson:"id,omitempty" json:"id,omitempty"`
	RequestedFileName   string             `bson:"requestedFileName,omitempty" json:"requestedFileName,omitempty"`
	Status              string             `bson:"status,omitempty" json:"status,omitempty"`
	ProgressPercentage  int                `bson:"progressPercentage,omitempty" json:"progressPercentage,omitempty"`
	CountOfMatchedFiles int                `bson:"countOfMatchedFiles,omitempty" json:"countOfMatchedFiles,omitempty"`
	CreateAt            string             `bson:"createAt,omitempty" json:"createAt,omitempty"`
	UpdateAt            string             `bson:"updateAt,omitempty" json:"updateAt,omitempty"`
	Result              string             `bson:"result,omitempty" json:"result,omitempty"`
}

File: nestjs-server/src/task/entities/task.entity.ts
import { AutoMap } from '@automapper/classes';
import { Column, Entity, ObjectID, ObjectIdColumn } from 'typeorm';
import { resultEnum } from '../enums/result.enum';
import { statusEnum } from '../enums/status.enum';

@Entity('tasks')
export class TaskEntity {

    @ObjectIdColumn()
    _id: ObjectID;

    @AutoMap()
    @Column()
    requestedFileName: string;

    @AutoMap()
    @Column()
    status: string = '';

    @AutoMap()
    @Column()
    progressPercentage: number = 0;

    @AutoMap()
    @Column()
    countOfMatchedFiles: number = 0;

    @AutoMap()
    @Column()
    createAt: string;

    @AutoMap()
    @Column()
    updateAt: string;

    @AutoMap()
    @Column()
    result: string = '';
}